src/Microsoft.FeatureManagement/FeatureDefinition.cs

```diff
@@ -42,5 +43,15 @@ public class FeatureDefinition
         /// A list of variant definitions that specify a configuration to return when assigned.
         /// </summary>
         public IEnumerable<VariantDefinition> Variants { get; set; } = Enumerable.Empty<VariantDefinition>();
+
+        /// <summary>
+        /// A flag to enable or disable sending telemetry events to the registered <see cref="ITelemetryPublisher">.
+        /// </summary>
+        public bool TelemetryEnabled { get; set; }
+
+        /// <summary>
+        /// A container for metadata relevant to telemetry.
+        /// </summary>
+        public IReadOnlyDictionary<string, string> TelemetryMetadata { get; set; }
```

Comment:
I know we meant metadata of a feature flag, but this name feels like metadata for telemetry to me and it's about telemetry configuration or something alike. Not sure if it's just me :) Is there another name, for example, TelemetryBag?

Reply:
Hmm I still prefer TelemetryMetadata. But here's some other options:
* TelemetryInfo
* TelemetryProperties
* TelemetryAttributes
* TelemetryContext
* TelemetryFields
* TelemetryDictionary

Reply:
TelemetryMetadata seems appropriate to me. Just like image metadata this is additional info related to the feature that is not essential to it's operation. The metadata is for telemetry purposes, so categorizing it as TelemetryMetadata seems reasonable.

Reply:
Sounds good. TelemetryMetadata it is then :)